lnwallet/rpcwallet/sign_coordinator.go

```diff
@@ -0,0 +1,886 @@
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) *signrpc.MuSig2SignResponse {
+			return resp.GetMuSig2SignResponse()
+		},
+	)
+}
+
+// SignMessage sends a SignMessageReq to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignMessage(_ context.Context,
+	in *signrpc.SignMessageReq,
+	_ ...grpc.CallOption) (*signrpc.SignMessageResp, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignMessageReq{
+		SignMessageReq: in,
+	}
+
+	return processRequest(
+		s, noTimeout,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) *signrpc.SignMessageResp {
+			return resp.GetSignMessageResp()
+		},
+	)
+}
+
+// SignPsbt sends a SignPsbtRequest to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignPsbt(_ context.Context,
+	in *walletrpc.SignPsbtRequest,
+	_ ...grpc.CallOption) (*walletrpc.SignPsbtResponse, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignPsbtRequest{
+		SignPsbtRequest: in,
+	}
+
+	return processRequest(
+		s, noTimeout,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) *walletrpc.SignPsbtResponse {
+			return resp.GetSignPsbtResponse()
+		},
+	)
+}
+
+// processRequest is a generic function that sends a request to the remote
+// signer and waits for the corresponding response. If a timeout is set, the
+// function will limit the execution time of the entire function to the
+// specified timeout. If it is not set, configured timeouts will be used for
+// the individual operations within the function.
+func processRequest[R comparable](s *SignCoordinator, timeout time.Duration,
+	generateRequest func(uint64) walletrpc.SignCoordinatorRequest,
+	extractResponse func(*signerResponse) R) (R, error) {
+
+	var zero R
+
+	done, err := s.registerRequest()
+	if err != nil {
+		return zero, err
+	}
+	defer done()
+
+	startTime := time.Now()
+
+	// If a timeout is enforced, we will wait for the connection using the
+	// specified timeout. Otherwise, we will wait for the connection using
+	// the configured connection timeout.
+	if timeout != 0 {
+		err = s.waitUntilConnectedWithTimeout(timeout)
+	} else {
+		err = s.WaitUntilConnected()
+	}
+
+	if err != nil {
+		return zero, err
+	}
+
+	reqID := s.nextRequestID.Add(1)
+	req := generateRequest(reqID)
+
+	cleanUpChannel := s.createResponseChannel(reqID)
+	defer cleanUpChannel()
+
+	log.Debugf("Sending a %t to the remote signer with request ID %d",
+		req.SignRequestType, reqID)
```

Comment:
Should we perhaps also log the _content_ of the request and response (on both the client and server) in trace mode? Can format them to JSON using a helper like this:

```
func formatProtoJSON(msg proto.Message) string {
	jsonBytes, err := lnrpc.ProtoJSONMarshalOpts.Marshal(msg)
	if err != nil {
		return fmt.Sprintf("<err: %v>", err.String())
	}
	return string(jsonBytes)
```

Reply:
Ah yeah, good idea! Added that as per your suggestion. Thanks 🙏